content/labs/lab10/index.en.md

```diff
@@ -0,0 +1,162 @@
+---
+title: "Lab10"
+weight: 10
+---
+
+# Lab 10: Parallel and Asynchronous Programming
+
+## Starter Code
+
+> [!NOTE]
+> **Student**
+> {{< filetree dir="labs/lab10/student" >}}
+
+## Generating Fractals
+
+{{% hint info %}}
+**What is parallel programming?**
+
+Parallel programming is a programming paradigm that utilizes the architecture of modern, multi-core processors (CPUs) to perform multiple computations simultaneously.
+
+Instead of processing data sequentially in a single thread, we divide the work among multiple threads that execute their tasks at the same time.
+
+{{% /hint %}}
+
+### Task Description
+
+The goal of the task is to implement and compare the execution time of different methods for parallelizing the computations performed while generating the [Mandelbrot Set](https://en.wikipedia.org/wiki/Mandelbrot_set).
+
+The starter code contains an abstract class `MandelbrotSetGenerator` that manages the entire process of generating the fractal and saving it to a `.png` file.
+
+The `SingleThreadGenerator` class is a concrete single-threaded implementation of the generator. It uses a simple, nested `for` loop to iterate over all pixels of the generated image. It will serve as a baseline for performance measurement.
+
+You must implement the following computation parallelization methods:
+
+- `MultiThreadGenerator`: A multi-threaded method that manually creates and manages `Thread` objects.
+- `TasksGenerator`: A method that parallelizes the work using the thread pool (`ThreadPool`) via `Task` objects.
+- `ParallelGenerator`: A method that uses the TPL (Task Parallel Library).
```

Comment:
```suggestion
- `ParallelGenerator`: A method that uses the `Parallel` class from TPL (Task Parallel Library).
```
TPL to trochę obszerniejszy temat. TPL zawiera ThreadPoola, Taski, Cancellation tokeny i między innymi właśnie klasę Parallel.